### ISSUE TITLE
Possible to control Sharepoint app access on a specific SharePoint site collection?

### ISSUE THREAD

**Issue by jdillard** [vgrem/Office365-REST-Python-Client#391]
Is it possible to use this newish Microsoft Graph workflow to control Sharepoint app access on a specific SharePoint site collection using this client?

https://developer.microsoft.com/en-us/office/blogs/controlling-app-access-on-specific-sharepoint-site-collections/

```
POST /sites/{id}/permissions/
```

https://docs.microsoft.com/en-us/graph/api/site-get-permission?view=graph-rest-1.0&tabs=http

**Comment by vgrem** [comment]
Greetings!

This is supported but _only in the latest version atm_ , here is an example:


```python
from office365.graph_client import GraphClient
from office365.onedrive.permissions.permission import Permission
from office365.onedrive.sites.site import Site

def acquire_token_by_client_credentials():
    authority_url = 'https://login.microsoftonline.com/{0}'.format(tenant)
    app = msal.ConfidentialClientApplication(
        authority=authority_url,
        client_id=client_id,
        client_credential=client_secret
    )
    return app.acquire_token_for_client(scopes=["https://graph.microsoft.com/.default"])

client = GraphClient(acquire_token_by_client_credentials)

site = client.sites.root  # type: Site
site_permission = site.permissions.add(
    roles=["write"],
    grantedToIdentities=[{
        "application": {
            "id": "89ea5c94-7736-4e25-95ad-3fa95f62b66e",
            "displayName": "Contoso Time Manager App"
        }
    }]
).execute_query()   # type: Permission
```


The latest version could be installed via GitHub::

`pip install git+https://github.com/vgrem/Office365-REST-Python-Client.git`

**Comment by vgrem** [comment]
Closing this one since this operation is included in a new version ([`2.3.7`  release](https://github.com/vgrem/Office365-REST-Python-Client/tree/2.3.7)) of library.

**Comment by jdillard** [comment]
Sorry, just getting around to testing this, thanks for making it available and adding the example. How do you specify the Site ID, `{id}` in the original post, that you would like to grant access to? It seems `root` is hardcoded in the example, when you should be able to specify a site:

```site = client.sites.root  # type: Site```
